### PR TITLE
2D UNet Hip MRI segmentation - s4809773

### PR DESCRIPTION
**Pull request by student480**
Hello Teaching Team, 

Here is my pull request for my chosen project of the 2D UNet segmentation of Hip MRI data of a prostate cancer study (the EASY level task). The files submitted in this pull request are as follows:

images_for_README/ folder containing all images used in the README file such as training loss graphs and validation images

README.md: markdown readme file detailing information about the model, its components, how it was trained, and its validation testing for accuracy

dataset.py: dataloader file used for converting nifti files into a form that the model can understand

modules.py: contains the model architecture - a largely standard UNet in this case

predict.py: takes in a saved model, and measures its performance against a validation set of data, additionally outputting images of its performance

train.py: file responsible for the training of the model, saving the model once trained, and producing a training loss graphic

Thank you,
Roman Kull s4809773

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.9222. 

**Model and functions:**

- It correctly uses PyTorch to construct the 2D-UNet models and functions.
- No data augmentation.
- Using original Train/Val datasets to train UNet model and observe the model performance. However, students are expected to split the train dataset into new train/val dataset to train model and observe the model performance in the training phase, then to use the original validation dataset to test the model performance on the unseen data.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- Good function docstrings

**Difficulty:** Easy.

**Additional Comments:**

- Minimal commits
- UNet introduction, User guide, conclusion, and reference are expected to be in the report.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. Minimal.</td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. No referencing.</td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). Submitted on time. </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-15</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Segmentations not one hot encoded -3
